app/controllers/hq/sponsorships_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Hq::SponsorshipsController < HqController
+
+  def inactivate
+    sponsorship = Sponsorship.find(params[:id])
+
+    @sponsorship_inactivator = InactivateSponsorship.new(sponsorship: sponsorship,
+                                                         end_date: params[:sponsorship][:end_date])
+    inactivate_sponsorship and redirect_to hq_sponsor_path(sponsorship.sponsor_id)
```

Comment:
Why in so many places we are hard-coding the `id` as the identifier? Rails is capable of handling `redirect_to orphan_path(@orphan)` or even `redirect_to @orphan` (http://guides.rubyonrails.org/routing.html#creating-paths-and-urls-from-objects). If we e.g. implement vanity URLs and override `to_param`, we'll have to go through all this code and convert `sponsorship.sponsor.id` to `sponsorship.sponsor.slug` or such.